### ISSUE TITLE
Constructor (1997) loading UNIVBE causes illegal write messages in the console

### ISSUE THREAD

**Issue by mckellenjaundice122** [dosbox-staging/dosbox-staging#3835]
### Are you using the latest Dosbox-Staging Version?

- [X] I have checked releases and am using the latest release.

### What Operating System are you using?

Windows 10

### Relevant hardware info

i5-7200U @ 2.50 GHz, 8 GB Ram, Intel HD 630

### Have you checked that no other similar issue already exists?

- [X] I have searched and not found similar issues.

### A clear and concise description of what the bug is.

When the game first starts, it launches its own UniVBE via a "univbe.drv" file which can by bypassed with a modified driver that bypasses this and cancels univbe from being loaded and fixes this bug, but oddly enough this issue seems to have cropped up a long time ago, disappeared and then showed up again looking through old forum. There's a mention of this as early back as 2004 on the dosbox compatibility page.

During the launching of univbe, dosbox hangs for a couple seconds and prints out a dozen "Illegal write to..." lines until the game actually starts. If the game is quit from this point and then launched again without restarting dosbox, this bug doesn't seem to happen. I have tried in both the latest version of staging 0.81.2, 0.80.1 and 0.76.0 all on a default .conf file and the bug remains. It also happens in mainline dosbox 0.74. I have already wrote about this on the game issues page on the staging wiki along with the fixed univbe.drv, but this might be something worth looking into.

![dosbox_6UJdmgWiWY](https://github.com/user-attachments/assets/54a9001d-02c3-4a41-92f2-864346c2702e)
The status window
![dosbox_YXjy620oV3](https://github.com/user-attachments/assets/affa6ceb-40f6-460d-a233-d2822ce5306f)
What it looks like in dosbox while the illegal write lines are being printed during its brief freeze

### Steps to reproduce the behaviour.

Launch constructor
The game will attempt to run its own univbe driver
The status window will be spammed with illegal writes while dosbox is frozen
Couple seconds later, the game eventually starts fine.

### Your configuration

```ini
This bug occurs with the default configuration
```

### Code of Conduct & Contributing Guidelines

- [X] Yes, I agree.

**Comment by Grounded0** [comment]
Constructor player here. The game itself works flawlessly and there's actually a method that was developed by a team member @LowLevelMahn called NoUNIVBE to work around this problem:

https://github.com/LowLevelMahn/NoUniVBE

This is something we'll won't fix and I'll close this in a couple of days. Thanks for writing on the Wiki about the patched UNIVBE driver that fixes this issue.

**Comment by mckellenjaundice122** [comment]
Do you happen to know when this bug started happening in dosbox? On my search i came across an ancient video from 2008 on youtube where it seems like this doesn't happen in 0.72, so I think this might be a regression that popped up sometime after that version of dosbox. Good to know it's not just me then, I wonder how the gog version gets past this?

**Comment by Grounded0** [comment]
We could start suppressing those messages and that would make the console less useful in finding out what's happening when there's a problem. When I was responsible for the Windows user experience the default setting was to run without the console exactly for these reasons. There's nothing there to see for your average user and I don't use the console either by default.

**Comment by johnnovak** [comment]
> Running DOSBox Staging without console is something I would recommend for normal users since you only need that information if you encounter a problem.

> There's nothing there to see for your average user and I don't have the console visible by default either.

Objection, how would the normal user know if there is a problem? Without the console, they'd be flying blind, with zero indication if they made a typo in the config file of it they're using invalid settings, etc.

That's the reason why I made opening the console the default and added the "Run with console" icons to the macOS DMG installer.

Until we have alternative ways to surface errors other than logging stuff to the console (e.g. OSD and printing errors to the DOS screen), having the console open by default is crucial. Some errors can trip up even advanced users, and you won't see helpful hints and deprecation warnings either. Moreover, some features become impossible to use, e.g. the joystick calibration that logs the settings you need to paste into your config.

Otherwise I agree, this is not a game issue and should be closed. Many demos and other programs trigger illegal write warnings and yeah, they probably point to some emulation error, but I don't think we have any special interest in making UNIVBE running flawlessly.

**Comment by Grounded0** [comment]
You could still test this on DOSBox-X does it trigger this console warning @mckellenjaundice122.

At least we could then dissect where we have an emulation deficiency.

**Comment by mckellenjaundice122** [comment]
I can confirm this doesn't happen in dosbox-x. There's a different "Error cpu: Illegal unhandled interrupt", but that shows up regardless of using the patched or original univbe.drv, so i think it's safe to say this specific bug doesn't exist in dosbox-x.
![dosbox-x_JptAsllHBr](https://github.com/user-attachments/assets/af12d488-c1db-40ca-92d2-0d194734f278)

**Comment by Grounded0** [comment]
We'll keep this open for now.

**Comment by interloper98** [comment]
I can't reproduce it.

@mckellenjaundice122 - can you do these two things:

1. Delete the stale UniVBE settings in the `construc/SETTINGS` directory. These are the files:
    - `UNIVBE.DRV`
    - `UVCONFIG.DAT`

    This is a one-time step.

2. Use the following `dosbox.conf`:

    ```ini
    [dosbox]
    memsize = 128
    
    [mouse]
    dos_mouse_immediate = true
    mouse_sensitivity = 200,120
    
    [cpu]
    core = dynamic
    cpu_cycles_protected = 30000
    
    [dos]
    xms = off
    ems = off
    umb = off
    
    [autoexec]
    mount c . -freesize 99999
    imgmount d "cd/CONSTRUCTR1.iso" -t cdrom
    c:\
    cd construc
    game
    ```
 
This fixes the sluggish and asymmetric mouse movement and the cycles are enough to maintain peak frame rates.

When I run this, there are no illegal writes or instruction issues in the logs.

**Comment by interloper98** [comment]
@Python-Exoproject :arrow_up: , just some suggestions to get a very pleasant experience in this game under DOSBox Staging.

**Comment by Grounded0** [comment]
Deleting `UNIVBE.DRV` and `UVCONFIG.DAT` forcing another lookup process indeed fixes the issue. Will close in a day.

**Comment by Grounded0** [comment]
Updated Wiki with the new fix.

**Comment by interloper98** [comment]
Thank you, @Grounded0!

**Comment by mckellenjaundice122** [comment]
Sorry for replying late, i don't check or use github too often. I tried both your config and also deleting the UVCONFIG and UNIVBE.drv and i still get the same error in console in the status window. I have no idea what i'm doing wrong, but since you've removed the modified univbe driver from the wiki which seems to be the only thing that works for me, i'll leave it here instead so others if they come across this issue can fix it this way if deleting the files doesn't work, at least for me it doesn't.

https://www.dosdays.co.uk/topics/graphics_drivers.php

**Comment by johnnovak** [comment]
> Sorry for replying late, i don't check or use github too often. I tried both your config and also deleting the UVCONFIG and UNIVBE.drv and i still get the same error in console in the status window. I have no idea what i'm doing wrong, but since you've removed the modified univbe driver from the wiki which seems to be the only thing that works for me, i'll leave it here instead so others if they come across this issue can fix it this way if deleting the files doesn't work, at least for me it doesn't.
> 
> https://www.dosdays.co.uk/topics/graphics_drivers.php

One possibility is you're using a different version of the game? Just guessing.

**Comment by mckellenjaundice122** [comment]
Hard to tell, you're probably right. I guess it's better off just leaving it like this, having two different type of fixes depending on how your game behaves. Saves everyone time i would assume if it's nothing critical, just good to have it documented at least.

Edit: This seems to be fixed completely as of 0.82.1. I have done additional tests, and if i allow the game to generate the UNIVBE driver using 0.80.1, it results in the illegal read error as seen above. Deleting the univbe files and doing it again in 0.80.1 just results in the error again, however, deleting the files and starting fresh in 0.82.1 has no issues.

Edit 2: I have been racking my brains for a while about this problem, just kind of lurking on the back of my mind, why this problem existed and why it seemed like i was the only one to have it for a while. Now that AIslop is finally ""good"" enough, at least better than me at analyzing files, it indeed seemed to have been a bug that originated with mainline 0.74 and carried over to Staging until it was eventually fixed, when i made this issue, i was still using 0.80.1, and using the .drv file it generated in 82.1, thus, the errors. Letting 0.82.1 generate the .drv and then using it either in mainline or any other older versions of staging will fix this. According to gemini (if it's to be trusted, since i don't have anyone else to ask or listen to me). This was because UNIVBE would register more memory than available to UNIVBE to use, thus the illegal addresses, this is already fixed by now in current staging versions, but it remains broken in mainline dosbox 0.74. 

Came across this little nasty issue with Terra Nova since it uses the same version of univbe and etc, so i encountered this again, hence the re-discovery, probably won't matter to most people.

Anyway, just thought it was interesting to share, gives me a bit of closure.

<img width="768" height="2241" alt="Image" src="https://github.com/user-attachments/assets/16485cf8-ea93-41bd-883f-67c6b496aede" />

If anyone's ever interested, here's the .drv files too:

[univbe bugged.zip](https://github.com/user-attachments/files/25157745/univbe.bugged.zip)
